xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VaapiEGL.cpp

```diff
@@ -445,8 +456,282 @@ bool CVaapiTexture::TestInteropHevc(VADisplay vaDpy, EGLDisplay eglDisplay)
+}
+
+GLuint CVaapi2Texture::GetTextureY()
+{
+  return m_y.glTexture;
+}
+
+GLuint CVaapi2Texture::GetTextureVU()
+{
+  return m_vu.glTexture;
+}
+
+CSizeInt CVaapi2Texture::GetTextureSize()
+{
+  return m_textureSize;
+}
+
+bool CVaapi2Texture::TestEsh(VADisplay vaDpy, EGLDisplay eglDisplay, std::uint32_t rtFormat, std::int32_t pixelFormat)
+{
+#if HAVE_VAEXPORTSURFACHEHANDLE
+  int width = 1920;
+  int height = 1080;
+
+  PFNEGLCREATEIMAGEKHRPROC eglCreateImageKHR = (PFNEGLCREATEIMAGEKHRPROC)eglGetProcAddress("eglCreateImageKHR");
+  PFNEGLDESTROYIMAGEKHRPROC eglDestroyImageKHR = (PFNEGLDESTROYIMAGEKHRPROC)eglGetProcAddress("eglDestroyImageKHR");
+  if (!eglCreateImageKHR || !eglDestroyImageKHR)
+  {
+    return false;
+  }
+
+  // create surfaces
+  VASurfaceID surface;
+  VAStatus status;
+
+  VASurfaceAttrib attribs = { };
+  attribs.flags = VA_SURFACE_ATTRIB_SETTABLE;
+  attribs.type = VASurfaceAttribPixelFormat;
+  attribs.value.type = VAGenericValueTypeInteger;
+  attribs.value.value.i = pixelFormat;
+
+  if (vaCreateSurfaces(vaDpy, rtFormat,
+                       width, height,
+                       &surface, 1, &attribs, 1) != VA_STATUS_SUCCESS)
+  {
+    return false;
+  }
+
+  // check interop
+  VADRMPRIMESurfaceDescriptor drmPrimeSurface;
+  status = vaExportSurfaceHandle(vaDpy, surface,
+                                 VA_SURFACE_ATTRIB_MEM_TYPE_DRM_PRIME_2,
+                                 VA_EXPORT_SURFACE_READ_ONLY | VA_EXPORT_SURFACE_SEPARATE_LAYERS,
+                                 &drmPrimeSurface);
+
+  bool result = false;
+
+  if (status == VA_STATUS_SUCCESS)
+  {
+    auto const& layer = drmPrimeSurface.layers[0];
+    auto const& object = drmPrimeSurface.objects[layer.object_index[0]];
+    EGLint attribs[] = {
+      EGL_LINUX_DRM_FOURCC_EXT, static_cast<EGLint> (drmPrimeSurface.layers[0].drm_format),
+      EGL_WIDTH, width,
+      EGL_HEIGHT, height,
+      EGL_DMA_BUF_PLANE0_FD_EXT, static_cast<EGLint> (object.fd),
+      EGL_DMA_BUF_PLANE0_OFFSET_EXT, static_cast<EGLint> (layer.offset[0]),
+      EGL_DMA_BUF_PLANE0_PITCH_EXT, static_cast<EGLint> (layer.pitch[0]),
+      EGL_NONE
+    };
+
+    EGLImageKHR eglImage = eglCreateImageKHR(eglDisplay,
+                                             EGL_NO_CONTEXT, EGL_LINUX_DMA_BUF_EXT, nullptr,
+                                             attribs);
+    if (eglImage)
+    {
+      eglDestroyImageKHR(eglDisplay, eglImage);
+      result = true;
+    }
+
+    for (int object = 0; object < drmPrimeSurface.num_objects; object++)
+    {
+      close(drmPrimeSurface.objects[object].fd);
+    }
+  }
+
+  vaDestroySurfaces(vaDpy, &surface, 1);
+
+  return result;
+#else
+  return false;
+#endif
+}
+
+
+void CVaapi2Texture::TestInterop(VADisplay vaDpy, EGLDisplay eglDisplay, bool &general, bool &hevc)
+{
+  general = false;
+  hevc = false;
+
+  int major_version, minor_version;
+  if (vaInitialize(vaDpy, &major_version, &minor_version) != VA_STATUS_SUCCESS)
```

Comment:
this has to be moved up a level or it will be callen on each configure

Reply:
Hm. I wonder, should we just keep an initialized VADisplay around in the win system instead of returning a new one in `GetVADisplay()` every time?

Reply:
yes, we can keep it in CVaapiProxy

Reply:
nope, see my other comment. VADisplay is passed in CVaapiRenderPicture.

Reply:
moved it to CVaapiProxy now (only for Wayland to test), because it still saves reinitializations each time you start a movie and you don't have to worry in TestInterop about whether you have to initialize the display or not. can move it back though.

Reply:
this is wrong, VADisplay is carried in VaapiPresentPicture. this can be used in configure

Reply:
VADisplay form VaapiPresentPicture is being used, see https://github.com/xbmc/xbmc/pull/13333/commits/015867047e3ae2b9b63ac343adecfd7dc5b329b6#diff-8869e60284b5a072aa8ffd8e1e7f0fb4R88